### PR TITLE
feat(sinon): add matcher.in

### PR DESCRIPTION
**Pull request by SimonSchick**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://sinonjs.org/releases/v7.2.2/matchers/#sinonmatchinarray
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/sinon/ts3.1/index.d.ts [hunk 1]
```diff
@@ -1441,6 +1441,10 @@ declare namespace Sinon {
          * Requires the value to be a Symbol.
          */
         symbol: SinonMatcher;
+        /**
+         * Requires the value to be in the specified array.
+         */
+        in(allowed: any[]): SinonMatcher;
```
Comment (43081j) [inline types/sinon/ts3.1/index.d.ts]: may be worth using `unknown[]`, though it wouldn't make a difference other than one less `any` :p

**Review by 43081j — APPROVED** [review]
LGTM 👍

File: types/sinon/ts3.1/index.d.ts (hunk 1 above)
Comment (SimonSchick) [reply types/sinon/ts3.1/index.d.ts]: For arguments you pass it's pretty much irrelevant as `any` and `unknown` would be equivalent in this case, if it was the return value I'd agree.

Comment (43081j) [reply types/sinon/ts3.1/index.d.ts]: Of course, as I had already mentioned it makes no difference but means one day we could have a no-any lint rule on which is nice. Just personal preference

**Comment by SimonSchick** [comment]
@johnnyreilly @DanielRosenwasser maybe? Been open for a while :)

**Comment by johnnyreilly** [comment]
Thanks!

**Pull request merged**